Scrabble/scrabble.py

```diff
@@ -0,0 +1,300 @@
+    num_appearances: int
+        Number of appearances of the letter that appears the most times in a
+
+    """
+    letter_with_most_appearances = None
+    num_appearances = 0
+    word = ""
+    for w in word_list:
+        letter_counts = [(letter, w.count(letter)) for letter in set(w)]
+        max_letter_and_count = max(letter_counts, key=lambda x: x[1])
+        most_common_letter, count = max_letter_and_count
+        if count > num_appearances:
+            num_appearances = count
+            letter_with_most_appearances = most_common_letter
+            word = w
+    return (word, letter_with_most_appearances, num_appearances)
+
+def find_longest_anagrams(word_list):
+    """
+    Finds the longest anagrams in a list of words, by sorting the words into
+    bins (a dictionary) by length and comparing the character counts of every
+    word in the bin with each other.
+
+    Parameters:
+    -----------
+    word_list: string
+        List to search for anagrams.
+
+    Returns:
+    --------
+    longest_anagrams: list of list of strings
+        Each inner list contains strings that are anagrams of each other.
+
+    """
+    word_dict = {}
+    longest_anagrams = None
+    longest_anagram_length = 0
+    for word in word_list:
+        length = len(word)
+        if not word_dict.get(length):
+            word_dict[length] = [word]
+        else:
+            word_dict[length].append(word)
+    for word in word_list:
+        length = len(word)
+        if length < longest_anagram_length:
+            continue
+        word_letter_count = {letter: word.count(letter) for letter in set(word)}
+        anagrams = [word]
+        word_dict[length].remove(word)
+        for w in word_dict[length]:
+            w_letter_count = {letter: w.count(letter) for letter in set(w)}
+            if word_letter_count == w_letter_count:
+                anagrams.append(w)
+        if len(anagrams) == 1:
+            continue
+        if length == longest_anagram_length:
+            longest_anagrams.append(anagrams)
+        else:
+            longest_anagrams = [anagrams]
+            longest_anagram_length = length
+    return longest_anagrams
+
+def find_longest_anagrams_the_smart_way(word_dict):
+    """
+    Finds the longest anagrams in a list of words by putting them into bins
+    by their sorted string form.  Returns the bins with the longest strings.
+
+    Parameters:
+    -----------
+    word_list: string
+        List to search for anagrams.
+
+    Returns:
+    --------
+    longest_anagrams: list of list of strings
+        Each inner list contains strings that are anagrams of each other.
+
+    """
+    word_dict = {}
+    longest_anagrams = []
+    anagram_length = 0
+    for word in word_list:
+        sorted_word = "".join(sorted(word))
+        if not word_dict.get(sorted_word):
+            word_dict[sorted_word] = [word]
+        else:
+            word_dict[sorted_word].append(word)
+    for key in word_dict:
```

Comment:
instead iterate over the key, value pairs to avoid the cumbersome syntax of word_dict[key]